# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/nsClipboardProxy.cpp
# Commit: 726458f976ff
# Full Hash: 726458f976fff2b05bb0eaf7e940cd7be06d75f5
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2022-08-03 03:42:48
# Regressor Bug: 1781129
# File Overlap Count: 1
# Description:
#   Bug 1781129 - Part 1: Use BigBuffer for IPCDataTransfer, r=edgar
#   
#   The IPCDataTransfer type is used to transfer Clipboard/Drag & Drop payloads
#   over IPC to allow them to be written to or read from the relevant system
#   interfaces. Previously, the system which was used was somewhat complex, and
# ==============================================================================

diff -r 1aa5ec22f47a -r 726458f976ff widget/nsClipboardProxy.cpp
--- a/widget/nsClipboardProxy.cpp	Tue Aug 02 17:08:59 2022 +0000
+++ b/widget/nsClipboardProxy.cpp	Tue Aug 02 17:15:41 2022 +0000
@@ -33,7 +33,7 @@
   nsCOMPtr<nsIPrincipal> requestingPrincipal =
       aTransferable->GetRequestingPrincipal();
   nsContentPolicyType contentPolicyType = aTransferable->GetContentPolicyType();
-  child->SendSetClipboard(ipcDataTransfer, isPrivateData,
+  child->SendSetClipboard(std::move(ipcDataTransfer), isPrivateData,
                           IPC::Principal(requestingPrincipal),
                           contentPolicyType, aWhichClipboard);
 
@@ -50,8 +50,7 @@
   ContentChild::GetSingleton()->SendGetClipboard(types, aWhichClipboard,
                                                  &dataTransfer);
   return nsContentUtils::IPCTransferableToTransferable(
-      dataTransfer, false /* aAddDataFlavor */, aTransferable,
-      ContentChild::GetSingleton());
+      dataTransfer, false /* aAddDataFlavor */, aTransferable);
 }
 
 NS_IMETHODIMP